### PR TITLE
MNT: Give the 'val' column a more interesting name

### PR DESCRIPTION
**Pull request by ericdill**
1. I have no idea how much stuff this is going to break... All the tests seem to pass (locally at least), so that's a good sign
2. @danielballan I'm not necessarily suggesting that this should be the final iteration, but I think this is an interesting discussion point.  'val' is a non-descriptive term and I think that the dataframe that results from the `bin_on` operation would be more informative if the 'val' column name reflected the downsampling mechanism. 
3. See this notebook for the output of the bin_on dataframe: http://nbviewer.ipython.org/gist/anonymous/a88b0410a54cb0a71fce

### CODE REVIEW

**Comment by danielballan** [comment]
Yes, I like this. The kindest thing you could say about "val" as a column name is that it was not very descriptive. But more realistically, it could be misleading.

All things being equal, it's convenient to have column names that are also valid Python identifiers. (For the uninitiated: you have use to `df['value: mean']`, whereas `df.val` and `df['val']` are both valid since `val` is a valid identifier.)

In this case I agree that clarity is paramount. `'value: mean'` is an improvement. What about just `'mean'`? Or the name of whatever function was used to downsample? Not for reasons of convenience, but in fact to be democratic and consistent. Perhaps the "value" of interest is not the mean, but one of the  of the other summary statistics.

**Comment by ericdill** [comment]
I had actually done `'mean'` initially, but then quickly realized how difficult it is going to be to programmatically grab the column that is the downsampled values, as opposed to the stats column.  

I hadn't appreciated how annoying it is going to be to grab 'value: mean'.  That's a pretty compelling point

Is the 'values' column ALWAYS the first column in the dataframe that results from a call to `bin_on`?  If so, then it will still be pretty easy to programmatically grab the column that is (usually) of primary interest

**Comment by danielballan** [comment]
Yes, it is always the first column. Select columns positionally is not quite ~~idiomatic pandas~~ pandorable, but it _is_ guaranteed to work. For programmatic purposes, it's fine; for examples, we should go a different way.

Having just suggested `'mean'`, I realize now that that would shadow `df.mean()` the sum of all the columns. If you use `'mean'` as a name, `df['mean']` is the column and `df.mean` is the method. Bad idea!

We'll find something.

File: dataportal/muxer/data_muxer.py [hunk 1]
```diff
@@ -462,7 +462,10 @@ def resample(self, time_labels, binning, interpolation=None, agg=None,
 
             # Start by using the first point in a bin. (If there are actually
             # multiple points, we will either overwrite or raise below.)
-            result[name]['val'] = first_point[name]
+            val_name = downsample
+            if val_name is None or val_name == 'None':
+                val_name = 'raw'
+            result[name][val_name] = first_point[name]
```
Comment (danielballan) [inline dataportal/muxer/data_muxer.py]: This lines above my comment here will have to go inside the if statement below, before `continue` (but `continue` should stay).

File: dataportal/muxer/data_muxer.py [hunk 2]
```diff
@@ -492,7 +495,7 @@ def resample(self, time_labels, binning, interpolation=None, agg=None,
                                                 index=safe_bins)
                 logger.debug("Interpolating to fill %d of %d empty bins in %s",
                              len(safe_bins), has_no_points[name].sum(), name)
-                result[name]['val'].fillna(interpolated_points, inplace=True)
+                result[name][val_name].fillna(interpolated_points, inplace=True)
 
             # Short-circuit if we are done.
             if np.all(~has_multiple_points[name]):
```
Comment (danielballan) [inline dataportal/muxer/data_muxer.py:504]: Here you need to built a result column in the case that there is no upsample rule or the upsampling rule wasn't needed.

**Comment by ericdill** [comment]
Todo: Implement these naming rules, as discussed with @danielballan 
1. val_raw: this is the column name if no upsampling or downsampling operations occurred
2. val_upsample: this is the column name if ONLY upsampling occurred
3. val_downsample: see above for downsampling
4. val_upsample_downsample: this is the column name if both upsampling and downsampling operations occurred

**Comment by ericdill** [comment]
Notebook based on commit 117dcce

http://nbviewer.ipython.org/gist/anonymous/cbc7fe4b6d39378abda2

I wonder if simply pasting a link to nbviewer is a legit version of `git fetch dan`?

**Comment by danielballan** [comment]
I can confirm that it is.

File: dataportal/muxer/data_muxer.py [hunk 3]
```diff
@@ -524,14 +536,16 @@ def resample(self, time_labels, binning, interpolation=None, agg=None,
...
-                    lambda x: np.min(np.asarray(x.dropna()), 0))
-            result[name]['val'].where(~has_multiple_points[name], downsampled,
-                                      inplace=True)
+                std_series = g.apply(lambda x: np.std(np.asarray(x.dropna()), 0))
+                max_series = g.apply(lambda x: np.max(np.asarray(x.dropna()), 0))
+                min_series = g.apply(lambda x: np.min(np.asarray(x.dropna()), 0))
+
+            val_col_series.where(~has_multiple_points[name], downsampled, inplace=True)
+            result[name][val_col_name] = val_col_series
+            result[name]['count'] = count_series
```
Comment (danielballan) [inline dataportal/muxer/data_muxer.py:545]: I am not convinced that making `count_series` (and so on) a separate variable and than assigning it to `result` at the end is an improvement. It is little/no effect of performance, but it adds code.

**Comment by danielballan** [comment]
Other than the line comment I left, which is a judgment call, this is good to merge.

File: dataportal/muxer/data_muxer.py [hunk 4]
```diff
@@ -460,15 +460,20 @@ def resample(self, time_labels, binning, interpolation=None, agg=None,
...
+            val_col_series = pd.Series(data=first_point[name])
 
             # Short-circuit if we are done.
             if np.all(has_one_point[name]):
+                val_col_name += '_raw'
+                result[name][val_col_name] = val_col_series
                 continue
 
-            result[name]['count'] = counts[name]
+            count_series = counts[name]
```
Comment (danielballan) [inline dataportal/muxer/data_muxer.py:476]: Why does this have to change?

File: dataportal/muxer/data_muxer.py (hunk 3 above)
Comment (ericdill) [inline dataportal/muxer/data_muxer.py:545]: The ONLY reason I made each element a series first and then assigned it to result at the end was to preserve the notion of "the first sub-column is the data column" that you had initially had.

File: dataportal/muxer/data_muxer.py (hunk 4 above)
Comment (ericdill) [inline dataportal/muxer/data_muxer.py:476]: I think my other comment addresses this

Comment (danielballan) [inline dataportal/muxer/data_muxer.py:476]: Ah, that makes sense.

File: dataportal/muxer/data_muxer.py [hunk 5]
```diff
@@ -492,12 +497,19 @@ def resample(self, time_labels, binning, interpolation=None, agg=None,
...
 
             # Short-circuit if we are done.
             if np.all(~has_multiple_points[name]):
+                result[name][val_col_name] = val_col_series
+                result[name]['count'] = count_series
                 continue
 
+            # append the downsample method to the value column name,
+            # but keep it short
+            val_col_name += '_{}'.format(six.text_type(downsample)[:4])
```
Comment (danielballan) [inline dataportal/muxer/data_muxer.py]: I would rather try not cutting off the name at 4 characters. Let's try using the full name until or unless it becomes a problem.

**Comment by ericdill** [comment]
@danielballan removed the 4 character limit on the column label. This should be good to go

**Pull request merged**